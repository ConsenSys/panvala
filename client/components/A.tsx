import styled from 'styled-components';
import { color, typography } from 'styled-system';

const A: any = styled.a`
  color: inherit;
  cursor: pointer;
  ${color};
  ${({ bold }) => bold && 'font-weight: bold'};
`;

export default A;
