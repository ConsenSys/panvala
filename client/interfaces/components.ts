import * as React from 'react';

export interface StatelessPage<P = {}> extends React.SFC<P> {
  getInitialProps?: (ctx: any) => Promise<P>;
}

export interface IParameterChangesObject {
  [key: string]: {
    oldValue: string;
    newValue: string;
  };
}

export interface IGovernanceSlateFormValues {
  email: string;
  title: string;
  firstName: string;
  lastName?: string;
  organization?: string;
  summary: string;
  recommendation: string;
  parameters: IParameterChangesObject;
  stake: string;
}
