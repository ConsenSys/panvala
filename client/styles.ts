import { Theme } from 'styled-system';

export const COLORS: any = {
  primary: '#59B6E6',
  text: '#343434',

  grey1: '#000000',
  grey2: '#343434',
  grey3: '#8E9EA6',
  grey4: '#C5D3D9',
  grey5: '#F0F5F6',
  grey6: '#FCFDFE',
  greyBorder: '#E5ECED',

  white: '#FFFFFF',

  blue1: '#1872A0',
  blue2: '#489FCC',
  blue3: '#DEF0FA',

  yellow1: 'rgba(255, 245, 2, 0.2)',
  yellow2: '#C6B72C',

  green1: '#E5FFEE',
  green2: '#31AE5C',

  red1: 'rgba(255, 147, 147, 0.25)',
  red2: '#EA6B6B',
};

export const BUTTON_COLORS: any = {
  default: COLORS.grey5,
  submit: COLORS.grey5,
  firstChoice: COLORS.green1,
  secondChoice: COLORS.yellow1,
  firstChoiceFg: COLORS.green2,
  secondChoiceFg: COLORS.yellow2,
};

export const colors: any = {
  black: '#000',
  grey: '#8E9EA6',
  greys: {
    dark: '#343434',
    border: '#E5ECED',
    light: '#F0F5F6',
    veryLight: '#FCFDFE',
    disabled: '#C5D3D9',
  },
  white: '#FFF',
  transparent: 'transparent',
  blue: '#489FCC',
  blues: {
    dark: '#1872A0',
    light: '#DEF0FA',
  },
  yellow: '#C6B72C',
  yellows: {
    light: 'rgba(255, 245, 2, 0.2)',
  },
  green: '#31AE5C',
  greens: {
    light: '#E5FFEE',
  },
  red: '#EA6B6B',
  reds: {
    dark: '#DE3333',
    light: 'rgba(255, 147, 147, 0.25)',
  },
};

export const theme: Theme = {
  // 640px, 832px, 1024px, 1440px
  breakpoints: ['40em', '52em', '64em', '90em'],
  space: [0, 4, 8, 16, 32, 64, 128, 256, 512],
  fontSizes: [12, 14, 16, 20, 24, 36, 48, 80, 96],
  fontWeights: [100, 200, 300, 400, 500, 600, 700, 800, 900],
  lineHeights: {
    solid: 1,
    title: 1.25,
    copy: 1.5,
  },
  fonts: {
    serif: 'Canela, athelas, georgia, times, serif',
    sansSerif:
      '-apple-system, BlinkMacSystemFont, roboto, "avenir next", avenir, "helvetica neue", helvetica, ubuntu, noto, "segoe ui", arial, sans-serif',
  },
  borders: [0, '1px solid', '2px solid', '4px solid', '8px solid', '16px solid', '32px solid'],
  radii: [0, 2, 4, 16, 9999, '100%'],
  shadows: ['0px 5px 20px rgba(0, 0, 0, 0.1)'],
  sizes: [16, 32, 64, 128, 256],
  colors,
};
