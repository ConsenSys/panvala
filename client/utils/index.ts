// Re-export Panvala utils to be used throughout
export { default as panvala_utils } from '../../packages/panvala-utils';
