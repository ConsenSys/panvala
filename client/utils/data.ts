import { IProposal, ISlate } from '../interfaces';
import { statuses } from './status';

export const proposalsArray: IProposal[] = [
  {
    id: 2,
    firstName: 'Alice',
    lastName: 'Doe',
    email: 'alice@email.com',
    title: 'K Semantics for Web Assembly',
    summary:
      'A proposal to make Ethereum safer by enabling developers to formally verify smart contracts that run on Web Assembly.',
    tokensRequested: '200000',
    category: 'GRANT',
    totalBudget: '100 for this. 200 for that.',
    otherFunding: 'nope!',
    projectPlan: 'plan',
    projectTimeline: 'timeline',
    teamBackgrounds: 'backgrounds',
    awardAddress: '0xd09cc3bc67e4294c4a446d8e4a2934a921410ed7',
  },
  {
    id: 3,
    firstName: 'Bob',
    lastName: 'Doe',
    email: 'bob@email.com',
    title: 'Registry Builder Vote Weighting',
    summary:
      'A proposal to make Ethereum safer by improving a reusable library for token-curated registries.',
    tokensRequested: '175000',
    category: 'GRANT',
    totalBudget: '100 for this. 200 for that.',
    otherFunding: 'nope!',
    projectPlan: 'plan',
    projectTimeline: 'timeline',
    teamBackgrounds: 'backgrounds',
    awardAddress: '0xd09cc3bc67e4294c4a446d8e4a2934a921410ed7',
  },
  {
    id: 4,
    firstName: 'Cat',
    lastName: 'Doe',
    email: 'cat@email.com',
    title: 'Tidbit Dynamic Oracles',
    summary: 'A proposal to make Ethereum safer by improving a reusable library for oracles.',
    tokensRequested: '150000',
    category: 'GRANT',
    totalBudget: '100 for this. 200 for that.',
    otherFunding: 'nope!',
    projectPlan: 'plan',
    projectTimeline: 'timeline',
    teamBackgrounds: 'backgrounds',
    awardAddress: '0xd09cc3bc67e4294c4a446d8e4a2934a921410ed7',
  },
  {
    id: 5,
    firstName: 'Dog',
    lastName: 'Doe',
    email: 'dog@email.com',
    title: 'Infrastructure Monitoring',
    summary:
      'A proposal to make Ethereum safer by monitoring high value systems and assets on Ethereum.',
    tokensRequested: '200000',
    category: 'GRANT',
    totalBudget: '100 for this. 200 for that.',
    otherFunding: 'nope!',
    projectPlan: 'plan',
    projectTimeline: 'timeline',
    teamBackgrounds: 'backgrounds',
    awardAddress: '0xd09cc3bc67e4294c4a446d8e4a2934a921410ed7',
  },
];

export const slatesArray: ISlate[] = [
  {
    id: 0,
    // metadataHash: 'QmRZxt2b1FVZPNqd8hsiykDL3TdBDeTSPX9Kv46HmX4Gx1',
    category: 'GRANT',
    status: statuses.PENDING_VOTE,
    deadline: 1539044131,
    title: 'Panvala Award Committee',
    description:
      'In tempus sem orci, eu auctor mi finibus eu. Nam mi risus, pretium ut laoreet fermentum, cursus nec mi. Nulla ...',
    owner: 'Guy Reid',
    ownerAddress: '0xd115bffabbdd893a6f7cea402e7338643ced44a6',
    organization: 'Panvala',
    incumbent: true,
    proposals: proposalsArray,
    requiredStake: '500',
  },
  {
    id: 1,
    // metadataHash: 'QmRZxt2b1FVZPNqd8hsiykDL3TdBDeTSPX9Kv46HmX4Gx2',
    category: 'GRANT',
    status: statuses.PENDING_TOKENS,
    deadline: 1539044131,
    title: 'Grant Awards',
    description:
      'In tempus sem orci, eu auctor mi finibus eu. Nam mi risus, pretium ut laoreet fermentum, cursus nec mi. Nulla sodales ultrices tellus sodales consectetur.Nullam id mattis velit.',
    owner: 'Marcus Berten',
    ownerAddress: '0xd115bffabbdd893a6f7cea402e7338643ced44a6',
    organization: 'Grant Awards Group',
    incumbent: false,
    proposals: proposalsArray,
    requiredStake: '500',
  },
  // {
  //   id: 'QmRZxt2b1FVZPNqd8hsiykDL3TdBDeTSPX9Kv46HmX4Gx3',
  //   category: 'GOVERNANCE',
  //   status: statuses.PENDING_TOKENS,
  //   deadline: false,
  //   title: 'Panvala Dev',
  //   description:
  //     'In tempus sem orci, eu auctor mi finibus eu. Nam mi risus, pretium ut laoreet fermentum, cursus nec mi. Nulla sodales ultrices tellus sodales consectetur.Nullam id mattis velit.',
  //   owner: 'Amber Gilbert',
  //   ownerAddress: '0xd115bffabbdd893a6f7cea402e7338643ced44a6',
  //   organization: 'Panvala',
  //   incumbent: false,
  //   proposals: proposalsArray,
  //   requiredStake: '500',
  // },
  // {
  //   id: 'QmRZxt2b1FVZPNqd8hsiykDL3TdBDeTSPX9Kv46HmX4Gx4',
  //   category: 'GOVERNANCE',
  //   status: statuses.PENDING_VOTE,
  //   deadline: 1539044231,
  //   title: 'Governance Change',
  //   description:
  //     'In tempus sem orci, eu auctor mi finibus eu. Nam mi risus, pretium ut laoreet fermentum, cursus nec mi. Nulla sodales ultrices tellus sodales consectetur.Nullam id mattis velit.',
  //   owner: 'Victoria Johnson',
  //   ownerAddress: '0xd115bffabbdd893a6f7cea402e7338643ced44a6',
  //   organization: 'Governance Group',
  //   incumbent: false,
  //   proposals: proposalsArray,
  //   requiredStake: '500',
  // },
];
