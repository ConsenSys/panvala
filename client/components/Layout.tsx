import * as React from 'react';
import Head from 'next/head';
import styled from 'styled-components';
import { ToastContainer, toast } from 'react-toastify';
import { format } from 'date-fns';
import { AxiosResponse } from 'axios';

import Header from './Header';
import { IProposal, ISlate, IAppContext } from '../interfaces';
import { getAllProposals, getAllSlates } from '../utils/api';
import { baseToConvertedUnits } from '../utils/format';

export const AppContext: React.Context<IAppContext> = React.createContext<IAppContext>({
  slates: [],
  proposals: [],
  currentBallot: {
    startDate: 0,
    votingOpenDate: 0,
    votingCloseDate: 0,
    finalityDate: 0,
  },
  notifications: [],
});

export const LayoutWrapper = styled.div`
  font-family: 'Roboto';
  min-height: 100vh;
  margin: 2em 10em;
`;

type IProps = {
  title: string;
  children: any;
};

export default class Layout extends React.Component<IProps, IAppContext> {
  readonly state: IAppContext = {
    slates: [],
    proposals: [],
    currentBallot: {
      startDate: 0,
      votingOpenDate: 0,
      votingCloseDate: 0,
      finalityDate: 0,
    },
    notifications: [{ action: 'Welcome to Panvala!' }],
    onNotify: () => this.handleNotify,
    onRefreshProposals: () => this.handleRefreshProposals(),
    onRefreshSlates: () => this.handleRefreshSlates(),
    onHandleNotification: (n: any) => this.handleNotification(n),
  };

  constructor(props: IProps) {
    super(props);
    this.handleNotify = this.handleNotify.bind(this);
    this.handleRefreshProposals = this.handleRefreshProposals.bind(this);
    this.handleRefreshSlates = this.handleRefreshSlates.bind(this);
    this.handleNotification = this.handleNotification.bind(this);
  }

  // runs once, onload
  async componentDidMount() {
    // const slatesFromIpfs: any[] = await Promise.all(slateMultihashes.map(mh => ipfsGetData(mh)));
    // const slatesWithMHs = slates.map((s, i) => ({ ...s, hash: slateMultihashes[i] }));
    let slates: ISlate[] = await this.handleGetAllSlates();
    const proposals: IProposal[] | AxiosResponse = await this.handleGetAllProposals();

    let proposalData: IProposal[] = [];
    // sort proposals by createdAt
    if (Array.isArray(proposals)) {
      const sortedProposals = proposals.sort((a: IProposal, b: IProposal) => {
        const timestampA = format(a.createdAt, 'x');
        const timestampB = format(b.createdAt, 'x');
        return parseInt(timestampA) - parseInt(timestampB);
      });
      proposalData = sortedProposals;
    }

    const oneWeekSeconds = 604800;
    // Epoch 3
    // beginning of week 1 (2/1)
    const epochStartDate = 1549040401; // gateKeeper.functions.currentBatchStart()
    // end of week 11 (4/19)
    const week11EndDate = epochStartDate + oneWeekSeconds * 11; // 1555689601
    // end of week 12 (4/26)
    const week12EndDate = week11EndDate + oneWeekSeconds;
    // end of week 13 (5/3)
    const week13EndDate = week12EndDate + oneWeekSeconds;

    // prettier-ignore
    this.setState({
      slates,
      proposals: proposalData,
      currentBallot: {
        startDate: epochStartDate,
        votingOpenDate: week11EndDate,
        votingCloseDate: week12EndDate,
        finalityDate: week13EndDate,
      },
    });
  }

  handleNotify(note: string, custom: string) {
    if (custom) {
      (toast as any)[custom](note);
    } else {
      toast(note);
    }
  }

  async handleGetAllProposals() {
    let proposals: IProposal[] | AxiosResponse = await getAllProposals();

    // convert tokensRequested: base -> human
    if (Array.isArray(proposals)) {
      proposals = proposals.map((p: any) => {
        p.tokensRequested = baseToConvertedUnits(p.tokensRequested, 18);
        return p;
      });
    } else {
      throw new Error('invalid proposals type -- AxiosResponse');
    }

    return proposals;
  }

  async handleRefreshProposals() {
    const proposals: IProposal[] = await this.handleGetAllProposals();
    return this.setState({ proposals });
  }

  async handleGetAllSlates() {
    const slates: ISlate[] | AxiosResponse = await getAllSlates();
    console.log('slates:', slates);
    if (Array.isArray(slates)) {
      return slates;
    }

    return [];
  }

  async handleRefreshSlates() {
    const slates: ISlate[] = await this.handleGetAllSlates();
    return this.setState({ slates });
  }

  handleNotification(notification: any) {
    this.setState({
      notifications: [...this.state.notifications, notification],
    });
  }

  render() {
    const { children, title }: IProps = this.props;
    console.log('Layout state:', this.state);

    return (
      <LayoutWrapper>
        <Head>
          <title>{title}</title>
        </Head>

        <Header notifications={this.state.notifications} />
        <AppContext.Provider value={this.state}>{children}</AppContext.Provider>
        <ToastContainer
          position="bottom-right"
          autoClose={8000}
          hideProgressBar={true}
          newestOnTop={false}
          rtl={false}
          draggable={false}
          closeOnClick
          pauseOnHover
        />
      </LayoutWrapper>
    );
  }
}
