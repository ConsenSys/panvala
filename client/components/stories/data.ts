import Router from 'next/router';
import range from 'lodash/range';
import uuid from 'uuid/v4';
import { utils } from 'ethers';
import { ballotDates } from '../../utils/voting';
import { ISlate, IProposal } from '../../interfaces';
import '../../globalStyles.css';

// https://github.com/zeit/next.js/issues/1827#issuecomment-323314141
(Router as any).router = {
  push: () => {},
};

export const epochStartDate = 1541178000; // Nov 2, 2018 17:00:00 GMT
export const currentBallot = ballotDates(epochStartDate);
export const baseBallot = {
  epochNumber: 0,
  startDate: 1541178000,
  initialSlateSubmissionDeadline: 1544504400,
  votingOpenDate: 1547830800,
  votingCloseDate: 1548435600,
  finalityDate: 1549040400,
  slateSubmissionDeadline: { GRANT: 0, GOVERNANCE: 0 },
};

export const proposals: IProposal[] = [
  {
    id: 0,
    category: 'GRANT',
    firstName: 'Amanda',
    lastName: 'Crypto',
    title: 'A great project',
    tokensRequested: '5000000000',
    summary:
      "Plasma is a family of protocols which allow individuals to easily deploy high-throughput, secure blockchains. A smart contract on Ethereum’s main chain can ensure that users’ funds are secure, even if the “plasma chain” acts fully maliciously. This eliminates the need for a trusted pegging mechanism like that of sidechains. Plasma chains are non-custodial, allowing the prioritization of scalability without sacrificing security. We’ve devised a new architecture for building Plasma apps on one generalized plasma chain. It establishes a clean separation between the plasma layer and the application layer. We will publish a generalized Plasma predicate contract framework, which allows for upgradeability and composability of plasma contracts. Since plasma research moves so quickly, we realized we needed to develop an architecture that allowed for maximal modularity, to prevent vast chunks of code from being thrown away with each new research discovery. With that framework, we will refactor our existing codebase for secure payments using predicates. We want to use the generalized Plasma research we did and put it to the test in our codebase refactor following the launch of our testnet on January 31. Dogfooding is a critical part of the work we do, as there's no point to open sourcing the codebase if it's not readable or easy to use.",
    awardAddress: '0xd115bffabbdd893a6f7cea402e7338643ced44a6',
    projectTimeline: 'timeline',
    teamBackgrounds: 'backgrounds',
  },
];

export const unstakedSlate: ISlate = {
  id: 0,
  category: 'GRANT',
  status: 0,
  owner: 'John Doe',
  recommender: '0xd115bffabbdd893a6f7cea402e7338643ced44a6',
  organization: 'Team Recommender',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer eu nibh molestie, auctor ligula a, faucibus ante. Morbi dapibus enim in vulputate congue. Mauris feugiat gravida nibh, sed pellentesque eros pellentesque eu. Sed rutrum vitae magna sed aliquet. Suspendisse facilisis vulputate lobortis. Vestibulum sed dolor eu mi molestie pharetra. Duis ut diam aliquam, molestie erat non, scelerisque ligula. Curabitur accumsan ipsum pellentesque posuere ornare. Sed vulputate cursus accumsan. Morbi efficitur dictum magna, a imperdiet mauris aliquet vitae.',
  proposals,
  requiredStake: utils.bigNumberify('300000000000000000000'),
  verifiedRecommender: false,
  incumbent: false,
  staker: '0x0000000000000000000000000000000000000000',
  epochNumber: 1,
};

export function makeBallot(options?: any, prototype?: any) {
  return {
    ...baseBallot,
    ...prototype,
    ...options,
  };
}

export function makeSlate(options: any, prototype?: ISlate) {
  return {
    ...unstakedSlate,
    ...prototype,
    ...options,
    id: uuid(),
  };
}

export function makeSlates(num: number, prototype: ISlate) {
  return range(1, num).map(() => makeSlate({}, prototype));
}

export const acceptedGrantSlate = makeSlate({
  verifiedRecommender: true,
  status: 3,
  epochNumber: 0,
});

export const rejectedGrantSlate = makeSlate(
  {
    status: 2,
  },
  acceptedGrantSlate
);

export const acceptedGovSlate = makeSlate(
  {
    category: 'GOVERNANCE',
    verifiedRecommender: true,
    status: 3,
  },
  acceptedGrantSlate
);

export const rejectedGovSlate = makeSlate(
  {
    status: 2,
  },
  acceptedGovSlate
);
