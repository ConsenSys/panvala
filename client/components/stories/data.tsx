import Router from 'next/router';
import { utils } from 'ethers';
import { ballotDates } from '../../utils/status';
import { ISlate, IProposal } from '../../interfaces';
import '../../globalStyles.css';

// https://github.com/zeit/next.js/issues/1827#issuecomment-323314141
(Router as any).router = {
  push: () => {},
  prefetch: () => {},
};

export const epochStartDate = 1549040401;
export const currentBallot = ballotDates(15499990);

export const proposals: IProposal[] = [
  {
    id: 0,
    firstName: 'Amanda',
    lastName: 'Crypto',
    title: 'A great project',
    tokensRequested: '5000000000',
    summary: 'All sorts of amazing things',
    awardAddress: '0xd115bffabbdd893a6f7cea402e7338643ced44a6',
    projectTimeline: 'timeline',
    teamBackgrounds: 'backgrounds',
  },
];

export const unstakedSlate: ISlate = {
  id: 0,
  category: 'GRANT',
  status: 0,
  deadline: currentBallot.votingOpenDate,
  title: 'Some slate',
  owner: 'John Doe',
  recommenderAddress: '0xd115bffabbdd893a6f7cea402e7338643ced44a6',
  organization: 'Team Recommender',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer eu nibh molestie, auctor ligula a, faucibus ante. Morbi dapibus enim in vulputate congue. Mauris feugiat gravida nibh, sed pellentesque eros pellentesque eu. Sed rutrum vitae magna sed aliquet. Suspendisse facilisis vulputate lobortis. Vestibulum sed dolor eu mi molestie pharetra. Duis ut diam aliquam, molestie erat non, scelerisque ligula. Curabitur accumsan ipsum pellentesque posuere ornare. Sed vulputate cursus accumsan. Morbi efficitur dictum magna, a imperdiet mauris aliquet vitae.',
  proposals,
  requiredStake: utils.bigNumberify('300000000000000000000'),
  verifiedRecommender: false,
};
