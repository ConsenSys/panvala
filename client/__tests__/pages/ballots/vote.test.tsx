import * as React from 'react';
import { render, fireEvent, waitForElement } from 'react-testing-library';
import 'jest-dom/extend-expect';

import Vote from '../../../pages/ballots/vote';
import { slatesArray } from '../../../utils/data';
import { AppContext } from '../../../components/Layout';
import { BUTTON_COLORS, COLORS } from '../../../styles';

const oneWeekSeconds = 604800;
const epochStartDate = 1549040401;
const week11EndDate = epochStartDate + oneWeekSeconds * 11;
const week12EndDate = week11EndDate + oneWeekSeconds;
const week13EndDate = week12EndDate + oneWeekSeconds;

const currentBallot = {
  startDate: epochStartDate,
  votingOpenDate: week11EndDate,
  votingCloseDate: week12EndDate,
  finalityDate: week13EndDate,
};

const setup: any = () => {
  const { getByText, getByTestId, container } = render(
    <AppContext.Provider value={{ slates: slatesArray, currentBallot }}>
      <Vote />
    </AppContext.Provider>
  );
  const firstChoiceButton = getByText('1st Choice');
  return {
    firstChoiceButton,
    getByTestId,
    getByText,
    container,
  };
};

describe('Ballots - vote', () => {
  test.skip('should render correctly Ballot slate voting component per snapshot', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });

  test('Ballots - vote', async () => {
    const { firstChoiceButton } = setup();

    // white background
    expect(firstChoiceButton).toHaveStyle(`
      background-color: ${COLORS.white}
    `);

    // click event / wait
    fireEvent.click(firstChoiceButton);
    const newButton = await waitForElement(() => firstChoiceButton);

    // green background
    expect(newButton).toHaveStyle(`
      background-color: ${BUTTON_COLORS.firstChoice}
    `);
  });
});
