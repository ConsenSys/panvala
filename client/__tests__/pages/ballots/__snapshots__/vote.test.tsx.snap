// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ballots - vote should render correctly Ballot slate voting component per snapshot 1`] = `
<div>
  <div>
    <div
      class="flex justify-end"
    >
      <div
        class="Deadline__Wrapper-sc-1a3aqlz-1 gtXKcM"
      >
        <div
          class="Tag__Wrapper-l3n6sk-1 ceKKQx"
        >
          <div
            class="Deadline__StyledDeadline-sc-1a3aqlz-0 jpSHwD Tag__StyledTag-l3n6sk-0 cZzgaW"
          >
            BALLOT OPENS 2019-04-19 AT 13:00 PM EST
          </div>
        </div>
      </div>
    </div>
    <div
      class="b f2 tc mv4"
    >
      Submit Vote
    </div>
    <div
      class="vote__BallotWrapper-sc-1g9yn2o-0 cSohui"
    >
      <div
        class="pa4"
      >
        <div
          class="SectionLabel__StyledSectionLabel-sc-1fzxsyz-0 hMXPaX"
        >
          GRANTS
        </div>
        <label
          class="Label__StyledLabel-sc-1pdmyea-0 fEfxuf"
          required=""
        >
          Select your first and second choice slate
          <span
            class="Required__StyledRequired-sc-1du3aep-0 icgBVx"
          >
             *
          </span>
        </label>
        <div
          class="flex flex-wrap mt3"
        >
          <div
            class="Card__Wrapper-h3ymxi-0 fRdLKy"
          >
            <div
              class="flex"
            >
              <div
                class="Tag__Wrapper-l3n6sk-1 ceKKQx"
              >
                <div
                  class="Tag__StyledTag-l3n6sk-0 cZzgaW"
                >
                  GRANT
                </div>
              </div>
              <div
                class="Tag__Wrapper-l3n6sk-1 ceKKQx"
              >
                <div
                  class="Tag__StyledTag-l3n6sk-0 eAsWWP"
                >
                  PENDING VOTE
                </div>
              </div>
            </div>
            <div
              class="Card__CardTitle-h3ymxi-1 bSTxPd"
            >
              Panvala Award Committee
            </div>
            <div
              class="Card__CardSubTitle-h3ymxi-2 kWUzrK"
            >
              4 Grants Included
            </div>
            <div
              class="Card__CardDescription-h3ymxi-3 dTqFuR"
            >
              In tempus sem orci, eu auctor mi finibus eu. Nam mi risus, pretium ut laoreet fermentum, cursus nec mi. Nulla ...
            </div>
            <div
              class="Card__CardUser-h3ymxi-4 liiyUA"
            >
              <div
                class="Card__CardAddress-h3ymxi-5 dGLzLS"
              >
                0x D115 BFFA BBDD 893A 6F7C EA40 2E73 3864 3CED 44A6
              </div>
            </div>
            <div
              class="Card__ViewSlateDetails-h3ymxi-7 cjMtgH"
            >
              View slate details
            </div>
            <div
              class="Form__Separator-naepsk-0 eYZnFo"
            />
            <div
              class="Card__ViewSlateDetails-h3ymxi-7 cjMtgH"
            >
              Grant Proposals:
            </div>
            <div
              class="Card__CardProposal-h3ymxi-8 fsABEt"
            >
              K Semantics for Web Assembly
            </div>
            <div
              class="Card__CardProposal-h3ymxi-8 fsABEt"
            >
              Registry Builder Vote Weighting
            </div>
            <div
              class="Card__CardProposal-h3ymxi-8 fsABEt"
            >
              Tidbit Dynamic Oracles
            </div>
            <div
              class="Card__CardProposal-h3ymxi-8 fsABEt"
            >
              Infrastructure Monitoring
            </div>
            <div
              class="Form__Separator-naepsk-0 eYZnFo"
            />
            <div
              class="Card__CardDescription-h3ymxi-3 dTqFuR"
            >
              Select an option
            </div>
            <div
              class="Card__ChoiceOptions-h3ymxi-6 gjfasN"
            >
              <div
                class="Button__Wrapper-sc-1vedn6s-2 jHiEij"
              >
                <button
                  class="Card__ChoiceButton-h3ymxi-9 VlzJp Button__StyledButton-sc-1vedn6s-0 lahyEh"
                  data-testid="first-choice"
                >
                  1st Choice
                </button>
              </div>
              <div
                class="Button__Wrapper-sc-1vedn6s-2 jHiEij"
              >
                <button
                  class="Card__ChoiceButton-h3ymxi-9 VlzJp Button__StyledButton-sc-1vedn6s-0 lahyEh"
                  data-testid="second-choice"
                >
                  2nd Choice
                </button>
              </div>
            </div>
          </div>
          <div
            class="Card__Wrapper-h3ymxi-0 fRdLKy"
          >
            <div
              class="flex"
            >
              <div
                class="Tag__Wrapper-l3n6sk-1 ceKKQx"
              >
                <div
                  class="Tag__StyledTag-l3n6sk-0 cZzgaW"
                >
                  GRANT
                </div>
              </div>
              <div
                class="Tag__Wrapper-l3n6sk-1 ceKKQx"
              >
                <div
                  class="Tag__StyledTag-l3n6sk-0 llOOnS"
                >
                  PENDING TOKENS
                </div>
              </div>
            </div>
            <div
              class="Card__CardTitle-h3ymxi-1 bSTxPd"
            >
              Grant Awards
            </div>
            <div
              class="Card__CardSubTitle-h3ymxi-2 kWUzrK"
            >
              4 Grants Included
            </div>
            <div
              class="Card__CardDescription-h3ymxi-3 dTqFuR"
            >
              In tempus sem orci, eu auctor mi finibus eu. Nam mi risus, pretium ut laoreet fermentum, cursus nec mi. Nulla sodales ultrices tellus sodales consectetur.Nullam id mattis velit.
            </div>
            <div
              class="Card__CardUser-h3ymxi-4 liiyUA"
            >
              <div
                class="Card__CardAddress-h3ymxi-5 dGLzLS"
              >
                0x D115 BFFA BBDD 893A 6F7C EA40 2E73 3864 3CED 44A6
              </div>
            </div>
            <div
              class="Card__ViewSlateDetails-h3ymxi-7 cjMtgH"
            >
              View slate details
            </div>
            <div
              class="Form__Separator-naepsk-0 eYZnFo"
            />
            <div
              class="Card__ViewSlateDetails-h3ymxi-7 cjMtgH"
            >
              Grant Proposals:
            </div>
            <div
              class="Card__CardProposal-h3ymxi-8 fsABEt"
            >
              K Semantics for Web Assembly
            </div>
            <div
              class="Card__CardProposal-h3ymxi-8 fsABEt"
            >
              Registry Builder Vote Weighting
            </div>
            <div
              class="Card__CardProposal-h3ymxi-8 fsABEt"
            >
              Tidbit Dynamic Oracles
            </div>
            <div
              class="Card__CardProposal-h3ymxi-8 fsABEt"
            >
              Infrastructure Monitoring
            </div>
            <div
              class="Form__Separator-naepsk-0 eYZnFo"
            />
            <div
              class="Card__CardDescription-h3ymxi-3 dTqFuR"
            >
              Select an option
            </div>
            <div
              class="Card__ChoiceOptions-h3ymxi-6 gjfasN"
            >
              <div
                class="Button__Wrapper-sc-1vedn6s-2 jHiEij"
              >
                <button
                  class="Card__ChoiceButton-h3ymxi-9 VlzJp Button__StyledButton-sc-1vedn6s-0 lahyEh"
                  data-testid="first-choice"
                >
                  1st Choice
                </button>
              </div>
              <div
                class="Button__Wrapper-sc-1vedn6s-2 jHiEij"
              >
                <button
                  class="Card__ChoiceButton-h3ymxi-9 VlzJp Button__StyledButton-sc-1vedn6s-0 lahyEh"
                  data-testid="second-choice"
                >
                  2nd Choice
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="vote__Separator-sc-1g9yn2o-1 gQIqRR"
      />
      <div
        class="flex flex-column pv4 ph4 items-end"
      >
        <div
          class="flex"
        >
          <div
            class="Button__Wrapper-sc-1vedn6s-2 jHiEij"
          >
            <button
              class="Button__LargeButton-sc-1vedn6s-1 lfgnkA"
            >
              Back
            </button>
          </div>
          <div
            class="Button__Wrapper-sc-1vedn6s-2 jHiEij"
          >
            <button
              class="Button__LargeButton-sc-1vedn6s-1 lCuqq"
              type="submit"
            >
              Confirm and Submit
            </button>
          </div>
        </div>
        <div
          class="f7 w5 tr mr3"
        >
          This will redirect to a seperate MetaMask window to confirm your transaction.
        </div>
      </div>
    </div>
  </div>
</div>
`;
